### PR TITLE
If MP-CP is enabled, make REST Client use it as ExecutorService, to allow for CP in filters

### LINKED ISSUE #8538: RequestScoped context is not propagated using rest-client and mutiny
**Describe the bug**
When using Mutiny, rest-client and context propagation, the request context is not propagated.

**Expected behavior**
The context is propagated.

**Actual behavior**
The following exception is thrown: 
ERROR [org.jbo.res.res.i18n] (pool-4-thread-2) RESTEASY002020: Unhandled asynchronous exception, sending back 500: javax.ws.rs.ProcessingException: javax.enterprise.context.ContextNotActiveException: interface javax.enterprise.context.RequestScoped

**To Reproduce**
A project to reproduce it is available [here](https://github.com/AmsterdamFilho/async-rest-client). It is based on the rest-client quick start guide. A ContainerRequestFilter was added to capture the authorisation header of every request and save it to a RequestScoped bean. Then, a ClientRequestFilter was added to insert this authorisation to the header of the external request using the RequestScoped bean. When executing the synchronous endpoint, it works just fine. But when using async call with mutiny, the exception is thrown.

1. Execute the project in using ./mvnw quarkus:dev;
2. Execute curl -w '\n' -X GET localhost:8080/country/name/greece --header "Authorization: 12345"
3. Verify a correct response is returned;
4. curl -w '\n' -X GET localhost:8080/country/name-uni/greece --header "Authorization: 12345"
5. Verify an exception is thrown.

**Environment:**
 - Output of `java -version`: OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+10)
 - Quarkus version: 1.3.2.Final

### PR DESCRIPTION
**Pull request by FroMage**
Fixes #8538

### CODE REVIEW

File: extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/registerprovider/EchoResource.java [hunk 1]
```diff
@@ -1,17 +1,108 @@
...
+        ManagedContext requestContext = Arc.container().requestContext();
+        Assertions.assertTrue(requestContext.isActive());
+        Assertions.assertNotNull(requestBean);
+        Assertions.assertNotEquals(uniqueNumber, requestBean.getUniqueNumber());
+        // should not have passed from call-client to here
+        Assertions.assertNull(ResteasyContext.getContextData(String.class));
+        // should not have passed from call-client to here
+        Assertions.assertNull(ResteasyContext.getContextData(Long.class));
+        return CompletableFuture.completedFuture("OK");
+    }
```
Comment (cescoffier) [inline extensions/rest-client/deployment/src/test/java/io/quarkus/restclient/registerprovider/EchoResource.java:107]: Maybe add a test with Uni?

**Review by cescoffier — APPROVED** [review]
Would that also work with Uni? The Uni support is based on the CS support AFAIK.

**Comment by mkouba** [comment]
CC @manovotn

**Review by manovotn — APPROVED** [review]
LGTM

**Comment by FroMage** [comment]
> Would that also work with Uni? The Uni support is based on the CS support AFAIK.

In theory Uni don't come into play because this is all done in the filters, which don't depend on the return type of the endpoint, but only that they are "async". All async execution uses the `ExecutorService`, so that should fix it for everyone.

This is really not about context for callbacks that the user registers on the async values returned by the client.

**Pull request merged**